subsys/bluetooth/host/gatt.c

```diff
@@ -5210,6 +5227,71 @@ int bt_gatt_store_ccc(uint8_t id, const bt_addr_le_t *addr)
 	return 0;
 }
 
+/** Struct used to store both the id and the random address of a device when replacing
+ * random addresses in the ccc attribute's cfg array with the device's id address after
+ * pairing complete.
+ */
+struct addr_match {
+	const bt_addr_le_t *private_addr;
+	const bt_addr_le_t *id_addr;
+};
+
+uint8_t convert_to_id_on_match(const struct bt_gatt_attr *attr,
+							   uint16_t handle, void *user_data)
+{
+	struct _bt_gatt_ccc *ccc;
+	struct addr_match *device_addresses = user_data;
```

Comment:
As I earlier suggested, the variable name can probably be shortened (I'm in particular bringing this up since it will allow you to merge some split lines later in the code). E.g. simply `match` would be perfectly readable to me at least.

Reply:
Used match.